src/meshapi/views/uisp_import.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiResponse, extend_schema, extend_schema_view
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAdminUser
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from meshapi.util.network_number import NETWORK_NUMBER_MAX, NETWORK_NUMBER_MIN
+from meshapi.util.uisp_import.fetch_uisp import get_uisp_devices, get_uisp_links
+from meshapi.util.uisp_import.sync_handlers import (
+    import_and_sync_uisp_devices,
+    import_and_sync_uisp_links,
+    sync_link_table_into_los_objects,
+)
+
+
+@extend_schema_view(
+    summary="Run the UISP Import job for a single Network Number.",
+    post=extend_schema(tags=["UISP Import"]),
+    responses={
+        "200": OpenApiResponse(
+            description="API is up and serving traffic",
+            response=OpenApiTypes.STR,
+        ),
+        "400": OpenApiResponse(
+            description="User provided invalid input in slug",
+            response=OpenApiTypes.STR,
+        ),
+        "500": OpenApiResponse(
+            description="Server error, probbaly a misconfigured environment variable.",
+            response=OpenApiTypes.STR,
+        ),
+    },
+)
+@api_view(["POST"])
+@permission_classes([IsAdminUser])
+def uisp_import_for_nn(request: Request, network_number: int) -> Response:
+    logging.info(f"Received uisp import request for NN{network_number}")
+    if not network_number:
+        status = 400
+        m = "Please provide a network number."
+        logging.error(m)
+        return Response({"detail": m}, status=status)
+
+    try:
+        target_nn = int(network_number)  # Because I apparently can't trust nobody
+        # Must be in valid range.
+        # NETWORK_NUMBER_MIN is set to something meshdb can assign new stuff, but we have lower NNs
+        # hence, I hardcode 1
+        FIRST_NN = 1
+        if target_nn < FIRST_NN or NETWORK_NUMBER_MAX < target_nn:
+            status = 400
+            m = "Network number is not in valid range."
+            logging.error(m)
+            return Response({"detail": m}, status=status)
+    except ValueError:
+        status = 400
+        m = f"Network Number must be an integer between {NETWORK_NUMBER_MIN} and {NETWORK_NUMBER_MAX}."
+        return Response({"detail": m}, status=status)
```

Comment:
IMO we could simplify this to just change it to try to fetch a `Node` object with that network number. It requires NN assignment to have happened already, but like, that's a feature not a bug

I think maybe the UISP import logic requires the node to exist anyway, so it's not like we're doing them any favors here. If there are devices associated with the NN they enter but no `Node` object, I think the UISP import is essentially a no-op